core/state/statedb.go

```diff
@@ -62,14 +62,16 @@ type StateDB struct {
 	db         Database
 	prefetcher *triePrefetcher
 	trie       Trie
+	noTrie     bool
 	hasher     crypto.KeccakState
 	snaps      *snapshot.Tree    // Nil if snapshot is not available
 	snap       snapshot.Snapshot // Nil if snapshot is not available
 
 	// originalRoot is the pre-state root, before any changes were made.
 	// It will be updated when the Commit is called.
 	originalRoot common.Hash
-
+	expectedRoot common.Hash // The state root in the block header
```

Comment:
I guess we dont need this `expectedRoot`? Because it is introduced in pipecommit feature, not related with fastnode actually

Reply:
the expected root should be set for we can not calculate the root in fast node